controllers/crds/cninode_controller.go

```diff
@@ -202,10 +202,10 @@ func (r *CNINodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r *CNINodeReconciler) waitTillCNINodeDeleted(ctx context.Context, nameSpacedCNINode types.NamespacedName) error {
 	oldCNINode := &v1alpha1.CNINode{}
 
-	ctx, cancel := context.WithTimeout(context.TODO(), time.Second*10)
+	ctx, cancel := context.WithTimeout(context.TODO(), time.Second*3)
 	defer cancel()
 
-	return wait.PollUntilContextCancel(ctx, time.Second, true, func(ctx context.Context) (bool, error) {
+	return wait.PollUntilContextCancel(ctx, 500*time.Millisecond, true, func(ctx context.Context) (bool, error) {
```

Comment:
nit: Why don't you use simpler option (`PollUntilContextTimeout`)?

Reply:
My bad, I misread that `PollUntilContextTimeout` was deprecated, have updated this in the latest commit.